0014-runtime-signing-tx-with-hardware-wallet.md

```diff
@@ -682,57 +701,70 @@ In this case the hardware wallet renders the following UI.
+`tx.call.body.data`. This is outside of the scope of this ADR and the **blind
+signing**, explicitly allowed by the user, is performed.
+
+We propose the following UI:
 
 <!-- markdownlint-disable line-length -->
 ```ledger
-| Review Encrypted > | < BLIND > | < Tx hash (1/1) > | < Public key (1/1) > | <  Nonce (1/1) > | <    Fee    > | < Gas limit > | <  Network  > | <  ParaTime > | <             > | <             |
-|   Transaction      |  SIGNING! |     <TX_HASH>     |     <PUBLIC_KEY>     |      <NONCE>     |  <SYM> <FEE>  |  <GAS LIMIT>  |   <NETWORK>   |   <RUNTIME>   |     APPROVE     |     REJECT    |
-|   (ParaTime)       |           |                   |                      |                  |               |               |               |               |                 |               |
+| Review Contract > | < BLIND > | < Tx hash (1/1) > | < Address (1/1) > | <   Amount    > | <     Fee     > | < Gas limit > | <  Network  > | <  ParaTime > | <            > | <             |
+|      Call         |  SIGNING! |     <TX_HASH>     |     <ADDRESS>     | <SYM> <AMOUNT>  |   <SYM> <FEE>   |  <GAS LIMIT>  |   <NETWORK>   |   <RUNTIME>   |    APPROVE     |     REJECT    |
+|   (ParaTime)      |           |                   |                   |                 |                 |               |               |               |                |               |
 ```
 <!-- markdownlint-enable line-length -->
 
-`TX_HASH` is a hex representation of sha256 checksum of `tx.call.data` field.
+`TX_HASH` is a hex representation of sha256 checksum of `tx.call.body.data`
+field.
 
-`PUBLIC_KEY` is a hex representation of the 32-byte `tx.call.pk` field.
+`ADDRESS` is a hex-encoded address of the smart contract.
 
-`NONCE` is a hex representation of the 15-byte `tx.call.nonce` field.
+### Signing encrypted runtime transactions
 
-### Signing EVM transactions
+Encrypted transactions (`tx.call.format == 1`) contain call data inside the
+[envelope's `data` field][runtime-sdk envelope] encrypted with Deoxys-II
+ephemeral key and X25519 key derivation.
 
-#### Creating smart contract
-
-`evm.Create` transaction will not be managed by the hardware wallet because the
-size of the EVM byte code may easily exceed the wallet's encrypted memory size.
-
-<!-- markdownlint-disable-next-line no-duplicate-header -->
-#### Calling smart contract
-
-In contrast to `contracts.Call`, `evm.Call` would require contract ABI in order
-to extract argument names which are stored in the RLP-encoded transaction on
-the blockchain. We do not support this, so only **blind signing** is performed
-which the user needs to enable first. The UI should be as follows:
+The hardware wallet is not expected to implement any of these decryption
+schemes, neither it is safe to share the ephemeral key with anyone. Instead,
+the user should enable **blind signing** and the hardware wallet should
+show the hash of the encrypted transaction, the public key and the nonce:
```

Comment:
How would the wallet be able to show the public key and nonce if it is only given the transaction hash?

Reply:
Oh I guess I got confused by "show the hash of the encrypted transaction". So you probably want to say "show the hash of the encrypted call data".